### PR TITLE
Fix master_test flake

### LINKED ISSUE #19141: unit test flake: pkg/master is flaky
This has just happened:

--- FAIL: TestGetServersToValidate (0.20s)
	<autogenerated>:31: 
                        
	Error Trace:	master_test.go:131
		
	Error:		Not equal: 3 (expected)
			        != 4 (actual)
		
	Messages:	unexpected server list: map[string]apiserver.Server{"controller-manager":apiserver.Server{Addr:"127.0.0.1", Port:10252, Path:"/healthz", EnableHTTPS:false, Validate:(apiserver.ValidatorFn)(nil)}, "scheduler":apiserver.Server{Addr:"127.0.0.1", Port:10251, Path:"/healthz", EnableHTTPS:false, Validate:(apiserver.ValidatorFn)(nil)}, "etcd-0":apiserver.Server{Addr:"", Port:4001, Path:"/health", EnableHTTPS:false, Validate:(apiserver.ValidatorFn)(0xb915a0)}, "etcd-1":apiserver.Server{Addr:"127.0.0.1", Port:49244, Path:"/health", EnableHTTPS:false, Validate:(apiserver.ValidatorFn)(0xb915a0)}}

https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/bd58b05f3eaea3d3b21df7d80f8b0a0c6c2ec87e/kubernetes-pull-test-unit-integration/7420/artifacts/?debugUI=DEVELOPERS

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix #19141

### CODE REVIEW

File: pkg/storage/etcd/testing/utils.go [hunk 1]
```diff
@@ -150,7 +168,12 @@ func NewEtcdTestClientServer(t *testing.T) *EtcdTestServer {
 	}
 	server.Client, err = etcd.New(cfg)
 	if err != nil {
-		t.Errorf("Unexpected Error in NewEtcdTestClientServer (%v)", err)
+		t.Errorf("Unexpected error in NewEtcdTestClientServer (%v)", err)
+		defer server.Terminate(t)
```
Comment (gmarek) [inline pkg/storage/etcd/testing/utils.go]: What's the reason for defer here?

Comment (wojtek-t) [inline pkg/storage/etcd/testing/utils.go]: consistency with what was before (6 lines below)

File: pkg/storage/etcd/testing/utils.go [hunk 2]
```diff
@@ -124,6 +126,22 @@ func (m *EtcdTestServer) launch(t *testing.T) error {
...
+// waitForEtcd wait until etcd is propagated correctly
+func (m *EtcdTestServer) waitUntilUp() error {
+	membersAPI := etcd.NewMembersAPI(m.Client)
+	for start := time.Now(); time.Since(start) < 5*time.Second; time.Sleep(10 * time.Millisecond) {
+		members, err := membersAPI.List(context.TODO())
+		if err != nil {
+			glog.Errorf("Error when getting etcd cluster members")
+			continue
+		}
+		if len(members) == 1 && len(members[0].ClientURLs) > 0 {
```
Comment (gmarek) [inline pkg/storage/etcd/testing/utils.go]: You probably want to return error different than timeout if there is one member but ClientURLs is empty.

File: pkg/storage/etcd/testing/utils.go (hunk 1 above)
Comment (gmarek) [inline pkg/storage/etcd/testing/utils.go]: Yeah - maybe we should remove it from both those places then?

File: pkg/storage/etcd/testing/utils.go (hunk 2 above)
Comment (wojtek-t) [inline pkg/storage/etcd/testing/utils.go]: I did it consciously - I think that if ClientURLs and there is one members (which is exactly what is happening at the beginning), I want to return a timeout.

File: pkg/storage/etcd/testing/utils.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/storage/etcd/testing/utils.go]: done.

**Comment by wojtek-t** [comment]
@gmarek - PTAL

**Comment by gmarek** [comment]
e2e is not requires as this only touches unit tests, but we need to wait for unit/integration tests to finish before applying the label...

**Comment by wojtek-t** [comment]
FAIL	k8s.io/kubernetes/pkg/storage/etcd	4.049s - #18928 

@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
pkg/storage & pkg/registry/generic/etcd failures - #18928 

@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**